adminSiteClient/DataInsightIndexPage.tsx

```diff
@@ -0,0 +1,457 @@
+import { useContext, useEffect, useMemo, useState } from "react"
+import * as React from "react"
+import { Button, Card, Flex, Input, Radio, Select, Space, Table } from "antd"
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome"
+import {
+    faCopy,
+    faPen,
+    faUpRightFromSquare,
+} from "@fortawesome/free-solid-svg-icons"
+import { faFigma } from "@fortawesome/free-brands-svg-icons"
+
+import { AdminLayout } from "./AdminLayout.js"
+import { Timeago } from "./Forms.js"
+import { ColumnsType } from "antd/es/table/InternalTable.js"
+import {
+    buildSearchWordsFromSearchString,
+    filterFunctionForSearchWords,
+    highlightFunctionForSearchWords,
+    SearchWord,
+} from "../adminShared/search.js"
+import { DbPlainTag, OwidGdocDataInsightIndexItem } from "@ourworldindata/types"
+import {
+    copyToClipboard,
+    dayjs,
+    RequiredBy,
+    startCase,
+} from "@ourworldindata/utils"
+import { CLOUDFLARE_IMAGES_URL } from "../settings/clientSettings.js"
+import { AdminAppContext } from "./AdminAppContext.js"
+
+type PublicationFilter = "all" | "published" | "scheduled" | "draft"
+type Layout = "list" | "gallery"
+
+const DEFAULT_PUBLICATION_FILTER: PublicationFilter = "all"
+const DEFAULT_LAYOUT: Layout = "list"
+
+const editIcon = <FontAwesomeIcon icon={faPen} size="sm" />
+const linkIcon = <FontAwesomeIcon icon={faUpRightFromSquare} size="sm" />
+const figmaIcon = <FontAwesomeIcon icon={faFigma} size="sm" />
+const copyIcon = <FontAwesomeIcon icon={faCopy} size="sm" />
+
+function createColumns(ctx: {
+    highlightFn: (
+        text: string | null | undefined
+    ) => React.ReactElement | string
+}): ColumnsType<OwidGdocDataInsightIndexItem> {
+    return [
+        {
+            title: "Preview",
+            key: "preview",
+            render: (_, dataInsight) =>
+                hasImage(dataInsight) ? (
+                    <>
+                        <img
+                            className="border"
+                            src={makePreviewImageSrc(dataInsight)}
+                            style={{ maxWidth: 200 }}
+                        />
+                        <Button
+                            type="text"
+                            size="small"
+                            icon={copyIcon}
+                            onClick={() =>
+                                copyToClipboard(dataInsight.image.filename)
+                            }
+                        >
+                            Copy filename
+                        </Button>
+                    </>
+                ) : undefined,
+        },
+        {
+            title: "Title",
+            dataIndex: "title",
+            key: "title",
+            width: 300,
+            render: (title) => ctx.highlightFn(title),
```

Comment:
Might be useful to make this a link if the DI's been published? IME authors/charlie are often looking for ways to quickly get links to things to use in different places

Reply:
Nice! Good idea. Never let an opportunity to turn text into a link pass by...